### PR TITLE
instrument CMS course settings page

### PR DESCRIPTION
**Pull request by thraxil**
quick and dirty profiling again on a view that we are having performance problems with in Tahoe. Same basic mechanism and approach as #242. See that PR for more details.

### CODE REVIEW

**Comment by thraxil** [comment]
[(2) Instrument CMS settings view](https://trello.com/c/OnePUMhL/2385-2-instrument-cms-settings-view)

**Comment by thraxil** [comment]
The jenkins build on this is fixed (for some reason, it failed to pull the repo from github) but it doesn't seem to be notifying github: http://107.178.245.150/job/edx-platform/162/

File: cms/djangoapps/contentstore/views/course.py [hunk 1]
```diff
@@ -1049,64 +1102,120 @@ def settings_handler(request, course_key_string):
...
+                        perflog['branches']['remove_entrance_exam'] = True
                         delete_entrance_exam(request, course_key)
+                        perflog['checkpoints'].append({
+                            'time': time.time(),
+                            'name': 'delete_entrance_exam()'})
 
                 # Perform the normal update workflow for the CourseDetails model
+                perflog['end'] = time.time()
+                perflog['total_duration'] = perflog['end'] - perflog['start']
+                log.info("PERFLOG %s" % json.dumps(perflog))
```
Comment (melvinsoft) [inline cms/djangoapps/contentstore/views/course.py]: @thraxil I think we need to add the fix for the datetime that we added in dashboard log. Converting the `perflog` to `str` before doing the JSON dump.

Comment (thraxil) [reply cms/djangoapps/contentstore/views/course.py]: That change is here: https://github.com/appsembler/edx-platform/pull/243/commits/3a3f39f131ed59942832b0f2bf95d96563935c1d

`end` and `start` are both floats returned from `time.time()` and don't need any special handling.

Comment (melvinsoft) [reply cms/djangoapps/contentstore/views/course.py]: @thraxil Sorry about that!

**Pull request merged**